src/analysis/processing/qgsalgorithmserviceareafromlayer.cpp

```diff
@@ -0,0 +1,269 @@
+/***************************************************************************
+                         qgsalgorithmserviceareafromlayer.cpp
+                         ---------------------
+    begin                : July 2018
+    copyright            : (C) 2018 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmserviceareafromlayer.h"
+
+#include "qgsgeometryutils.h"
+#include "qgsgraphanalyzer.h"
+
+///@cond PRIVATE
+
+QString QgsServiceAreaFromLayerAlgorithm::name() const
+{
+  return QStringLiteral( "serviceareafromlayer" );
+}
+
+QString QgsServiceAreaFromLayerAlgorithm::displayName() const
+{
+  return QObject::tr( "Service area (from layer)" );
+}
+
+QStringList QgsServiceAreaFromLayerAlgorithm::tags() const
+{
+  return QObject::tr( "network,service,area,shortest,fastest" ).split( ',' );
+}
+
+QString QgsServiceAreaFromLayerAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm computes service areas from the set of start points defined by point layer using travel time or distance." );
```

Comment:
There was more detail in the original yaml file, why was this removed?

Reply:
Seems they were added *after* initial port. Anyway I copied them now.